### PR TITLE
Remove dual mode support

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**
Closes [#11545](https://github.com/crate/crate/issues/11545)

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/admin/auth/hba.rst [hunk 1]
```diff
@@ -200,46 +200,3 @@ For more information, consult the :ref:`privileges section
 
     cr> DROP USER admin;
     DROP OK, 1 row affected (... sec)
-
-
-.. _admin_hba_node:
-
-Node-to-node communication
-==========================
```
Comment (mfussenegger) [inline docs/admin/auth/hba.rst:208]: I think we should still keep the documentation but remove the note about ssl-mixed note.
It is still important to activate the client certificate check if you're communicating across trusted network borders.

Comment (BaurzhanSakhariev) [reply docs/admin/auth/hba.rst]: restored this section, removed only sentences about dual mode and ssl-mixed

File: docs/admin/auth/hba.rst [hunk 2]
```diff
@@ -228,14 +228,6 @@ this:
             ssl: on
             method: cert
 
-Here, communication between nodes in the same zone (i.e., matching the
```
Comment (mfussenegger) [inline docs/admin/auth/hba.rst:231]: I'd also change the sentence above from

```
For example, if you wanted to configure a `multi-zone cluster`_, you could do
this:
```

to 

```
For example, if you wanted to configure a `multi-zone cluster`_, you should enable certificate authentication like 
this:
```

and just have a single entry:


```
    auth:
      host_based:
        enabled: true
        config:
          0:
            protocol: transport
            ssl: on
            method: cert
```

**Review by mfussenegger — APPROVED** [review]

**Comment by mfussenegger** [comment]
@BaurzhanSakhariev could you squash the commits before merging?

**Pull request merged**